falcon/middleware.py

```diff
@@ -19,7 +22,7 @@ class CORSMiddleware(object):
 
     Keyword Arguments:
         allow_origins (Union[str, Iterable[str]]): List of origins to allow (case
-            sensitive). The string ``'*'`` acts as a wildcard, matching every origin.
+            sensitive) or callable. The string ``'*'`` acts as a wildcard, matching every origin.
```

Comment:
I think we'll need an update to `docs/api/cors.rst` as well to better explain how to use the callable option, provide an example.